Controllers/GemController.cs

```diff
@@ -9,16 +9,24 @@ namespace GemManager.Controllers
     public class GemController : ControllerBase
     {
         private readonly IGemRepository _gemRepository;
+        private readonly IRepository<User> _userRepository;
 
-        public GemController(IGemRepository gemRepository)
+        public GemController(IGemRepository gemRepository, IRepository<User> userRepository)
```

Comment:
It's a bit strange to give the gem controller, responsibility of both user repository and its own repository. I would suggest maybe removing the user repository from here, Gem has a reference to what user you want through its "from" parameter. Use that reference instead, in order to do your validation.